### PR TITLE
Basic dialog functionality when access to a watch item is denied

### PR DESCRIPTION
**Pull request by mlw**
Note: This is a work in progress. PR is being cut in the current state to help keep the review smaller. All policies are hard coded to be silent and nothing will be displayed for now. But, if the `silent` member of `WatchItemPolicy` were changed to `false` then a dialog would be displayed as expected.

### CODE REVIEW

File: Source/gui/SNTFileAccessMessageWindowController.m [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  [super windowWillClose:notification];
+}
+
+- (NSAttributedString *)attributedCustomMessage {
+  return [SNTBlockMessage formatMessage:self.customMessage];
+}
+
+- (NSString *)messageHash {
+  return [NSString stringWithFormat:@"%@|%@|%d", self.event.ruleName, self.event.ruleVersion,
+                                    [self.event.pid intValue]];
```
Comment (pmarkowsky) [inline Source/gui/SNTFileAccessMessageWindowController.m:74]: Does this overfit to a specific process shouldn't that be name or another ID that's more stable?

File: Source/gui/SNTFileAccessMessageWindowView.swift [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    }.frame(maxWidth:800.0).fixedSize()
+  }
+
+  func publisherInfo() {
+    print("showing publisher popup...")
+  }
+
+  func openButton() {
+    print("opening event info...")
+  }
```
Comment (pmarkowsky) [inline Source/gui/SNTFileAccessMessageWindowView.swift:122]: This probably needs the URL extension to the sync protocol to redirect uses to the right place.

File: Source/gui/SNTFileAccessMessageWindowView.swift [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+      Spacer()
+
+    }.frame(maxWidth:800.0).fixedSize()
+  }
+
+  func publisherInfo() {
+    print("showing publisher popup...")
+  }
+
```
Comment (pmarkowsky) [inline Source/gui/SNTFileAccessMessageWindowView.swift:119]: Please add a TODO to wire up cert info from the process.

File: Source/gui/SNTNotificationManager.h [hunk 4]
```diff
@@ -17,6 +17,7 @@
 #import "Source/common/SNTXPCNotifierInterface.h"
 #import "Source/gui/SNTBinaryMessageWindowController.h"
 #import "Source/gui/SNTDeviceMessageWindowController.h"
+#import "Source/gui/SNTFileAccessMessageWindowController.h"
```
Comment (pmarkowsky) [inline Source/gui/SNTNotificationManager.h:20]: This include probably don't need this in the header just the `.m`

File: Source/santad/EventProviders/SNTEndpointSecurityFileAccessAuthorizer.mm [hunk 5]
```diff
@@ -481,6 +485,27 @@ - (FileAccessPolicyDecision)handleMessage:(const Message &)msg
                               targetPathCopy, policyDecision);
                           }];
     }
+
+    if (!optionalPolicy.value()->silent && self.fileAccessBlockCallback) {
```
Comment (pmarkowsky) [inline Source/santad/EventProviders/SNTEndpointSecurityFileAccessAuthorizer.mm:489]: nit: This will always be false.

**Review by pmarkowsky — APPROVED** [review]
Overall looks good. I think this is ok to approve as we know there more PRs to follow and this leaves the features inert and keeps the size of the PR more manageable.

File: Source/gui/SNTFileAccessMessageWindowView.swift (hunk 2 above)
Comment (mlw) [reply Source/gui/SNTFileAccessMessageWindowView.swift:122]: Added TODO

File: Source/santad/EventProviders/SNTEndpointSecurityFileAccessAuthorizer.mm (hunk 5 above)
Comment (mlw) [reply Source/santad/EventProviders/SNTEndpointSecurityFileAccessAuthorizer.mm:489]: As discussed, on purpose for now as noted in `WatchItemPolicy.h`

File: Source/gui/SNTFileAccessMessageWindowController.m (hunk 1 above)
Comment (pmarkowsky) [reply Source/gui/SNTFileAccessMessageWindowController.m:76]: I opened #1109 to track this.

**Review by pmarkowsky — APPROVED** [review]
Reapproving.

**Pull request merged**